lib/viewmodels/profile_view_model.dart

```diff
@@ -1,21 +1,52 @@
 import "package:hooks_riverpod/hooks_riverpod.dart";
 import "package:proxima/models/ui/profile_data.dart";
+import "package:proxima/services/database/user_repository_service.dart";
 import "package:proxima/viewmodels/login_view_model.dart";
 
 /// User profile view model
 class ProfileViewModel extends AsyncNotifier<ProfileData> {
   ProfileViewModel();
 
-  @override
-  Future<ProfileData> build() async {
+  Future<ProfileData> _fetchData() async {
     final user = ref.watch(userProvider).valueOrNull;
+    final userDataBase = ref.watch(userRepositoryProvider);
+    final uid = ref.watch(uidProvider);
+
     if (user == null) {
       return Future.error(
         "User must be logged in before displaying the home page.",
       );
     }
+    if (uid == null) {
+      return Future.error(
+        "User id was not found.",
+      );
+    }
+    final userData = await userDataBase.getUser(uid);
+
+    return ProfileData(loginUser: user, firestoreUser: userData);
+  }
+
+  @override
+  Future<ProfileData> build() async {
+    return _fetchData();
```

Comment:
If the goal here is to refresh the state by recalling the build method, the code should use another system. See `ref.invalidateSelf();` and https://riverpod.dev/docs/essentials/side_effects#using-refinvalidateself-to-refresh-the-provider.